qiskit_nature/circuit/library/ansatzes/succ_full.py

```diff
@@ -0,0 +1,208 @@
+        excitations.extend(
+            generate_fermionic_excitations(
+                1,
+                num_spin_orbitals,
+                num_particles,
+                alpha_spin=self.include_singles[0],
+                beta_spin=self.include_singles[1],
+            )
+        )
+
+        num_electrons = num_particles[0]
+
+        # generate alpha-spin and beta-spin orbital indices for occupied and unoccupied ones
+        alpha_excitations = get_alpha_excitations(
+            num_electrons, num_spin_orbitals, self._generalized
+        )
+        beta_excitations = get_beta_excitations(num_electrons, num_spin_orbitals, self._generalized)
+        logger.debug("Generated list of single alpha excitations: %s", alpha_excitations)
+        logger.debug("Generated list of single beta excitations: %s", beta_excitations)
+
+        # Find all possible double excitations constructed from the list of single excitations.
+        # Note, that we use `product` here, in order to also get those double
+        # excitations form an alpha excitation and a beta excitation. We will need those in the
+        # following post-processing step.
+        pool = itertools.product(alpha_excitations, beta_excitations)
+
+        for exc in pool:
+            # find the two excitations (Note: SUCCD only works for double excitations!)
+            alpha_exc, second_exc = exc[0], exc[1]
+            # shift the second excitation into the beta-spin orbital index range
+            beta_exc = (
+                second_exc[0],
+                second_exc[1],
+            )
+            # add the excitation tuple
+            occ: Tuple[int, ...]
+            unocc: Tuple[int, ...]
+            occ, unocc = zip(alpha_exc, beta_exc)
+            exc_tuple = (occ, unocc)
+            excitations.append(exc_tuple)
+            logger.debug("Added the excitation: %s", exc_tuple)
+
+        return excitations
+
+    def _validate_num_particles(self, num_particles):
+        try:
+            assert num_particles[0] == num_particles[1]
+        except AssertionError as exc:
+            raise QiskitNatureError(
+                "The SUCC_full Ansatz only works for singlet-spin systems. However, you specified "
+                "differing numbers of alpha and beta electrons:",
+                str(num_particles),
+            ) from exc
+
+    def _build_fermionic_excitation_ops(self, excitations: Sequence) -> List[FermionicOp]:
+        """Builds all possible excitation operators with the given number of excitations for the
+        specified number of particles distributed in the number of orbitals.
+        Args:
+            excitations: the list of excitations.
+        Returns:
+            The list of excitation operators in the second quantized formalism.
+        """
+        operators = []
+        excitations_dictionary: Dict[int, List[Tuple[Tuple[int, ...], Tuple[int, ...]]]] = {}
+        for exc in excitations:
+            exc_level = sum(exc[1])
```

Comment:
I think this is the key aspect of this PR. I think this would be worth an explanatory comment and we may be able to find a better name for this variable. The idea here is, that `sum(exc[1])` sums over the indices of the virtual orbitals in which we excite into. Thus, the level of an excitations is indicated by this sum.
Maybe add a comment explaning this.